libindy/src/services/wallet/mod.rs

```diff
@@ -534,6 +500,43 @@ impl WalletService {
         res
     }
 
+    fn _prepare_metadata(&self, key: &str, key_derivation_method: &KeyDerivationMethod, keys: &Keys) -> Result<Vec<u8>, WalletError> {
+        let (master_key_salt, master_key) = match key_derivation_method {
+            KeyDerivationMethod::RAW =>
+                (None, encryption::raw_master_key(key)?),
+            KeyDerivationMethod::ARAGON2I_INT | KeyDerivationMethod::ARAGON2I_MOD => {
+                let master_key_salt = encryption::gen_master_key_salt()?;
+                let master_key = encryption::derive_master_key(key, &master_key_salt, key_derivation_method)?;
+                (Some(master_key_salt[..].to_vec()), master_key)
+            }
+        };
+
+        let metadata = Metadata {
```

Comment:
Can we make metadata enum based in key deriviation method in a backward compatible way?